server/controllers/uploadImage.js

```diff
@@ -0,0 +1,52 @@
+// @route POST /uploadImage/upload
+// @desc upload single image
+// @access Public
+
+const asyncHandler = require("express-async-handler");
+const { dataUri } = require("../middleware/multer");
+const { cloudinary } = require("../utils/cloudinary");
+
+exports.uploadImage = asyncHandler(async (req, res, next) => {
+  try {
+    const file = dataUri(req.file).content;
+    const uploadResponse = await cloudinary.uploader.upload(file);
```

Comment:
We'd want to somehow attach this to the user - so setting the file name to their id will work. Also gets rid of stale images in our CDN.